contrib/golang/filters/http/test/golang_filter_test.cc

```diff
@@ -0,0 +1,171 @@
+#include <cstdint>
+#include <memory>
+
+#include "envoy/config/core/v3/base.pb.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/http/message_impl.h"
+#include "source/common/stream_info/stream_info_impl.h"
+
+#include "test/common/stats/stat_test_utility.h"
+#include "test/mocks/api/mocks.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/server/factory_context.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+#include "test/mocks/upstream/cluster_manager.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/logging.h"
+#include "test/test_common/printers.h"
+#include "test/test_common/test_runtime.h"
+#include "test/test_common/utility.h"
+
+#include "absl/strings/str_format.h"
+#include "contrib/golang/filters/http/source/golang_filter.h"
+#include "gmock/gmock.h"
+
+using testing::_;
+using testing::AtLeast;
+using testing::InSequence;
+using testing::Invoke;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Golang {
+namespace {
+
+class TestFilter : public Filter {
+public:
+  using Filter::Filter;
+
+  // TODO: for mock methods
+  // MOCK_METHOD;
```

Comment:
? remove. Audit for all commented out code.

Reply:
Done. Thanks for your comments.